CMakeLists.txt

```diff
@@ -1,120 +1,111 @@
-    set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} UrlLib)
+if(JSRUNTIMEHOST_POLYFILL_XMLHTTPREQUEST)
+    FetchContent_MakeAvailable_With_Message(UrlLib)
+    set_property(TARGET UrlLib PROPERTY FOLDER Dependencies)
 endif()
 
 if(NAPI_JAVASCRIPT_ENGINE STREQUAL "V8" AND JSRUNTIMEHOST_CORE_APPRUNTIME_V8_INSPECTOR)
-    if(NOT TARGET asio)
-        set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} asio)
-    endif()
-
-    if(NOT TARGET llhttp_static)
-        set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "")
-        set(BUILD_STATIC_LIBS ON CACHE INTERNAL "")
-        set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} llhttp)
-    endif()
+    FetchContent_MakeAvailable_With_Message(asio)
+    add_library(asio INTERFACE)
+    target_include_directories(asio INTERFACE "${asio_SOURCE_DIR}/asio/include")
+    set_property(TARGET asio PROPERTY FOLDER Dependencies)
+
+    set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "")
+    set(BUILD_STATIC_LIBS ON CACHE INTERNAL "")
+    FetchContent_MakeAvailable_With_Message(llhttp)
+    set_property(TARGET llhttp_static PROPERTY FOLDER Dependencies)
 endif()
 
-# Fetching content
-message(STATUS "Fetching dependencies for ${PROJECT_NAME} (${CONTENT_TO_FETCH})")
-FetchContent_MakeAvailable(${CONTENT_TO_FETCH})
-message(STATUS "Fetching dependencies for ${PROJECT_NAME} - done")
-
-# Set properties after fetching content
-if(TARGET arcana)
-    set_property(TARGET arcana PROPERTY FOLDER Dependencies)
-endif()
+if(JSRUNTIMEHOST_TESTS AND (WIN32 OR (APPLE AND NOT IOS) OR (UNIX AND NOT ANDROID)))
+    if(WIN32)
+        # For Windows: Prevent overriding the parent project's compiler/linker settings
+        # Default build type for my test projects are /MDd (MultiThreaded DLL) but GTests default to /MTd (MultiTreaded)
+        # see https://github.com/google/googletest/blob/main/googletest/README.md
+        # "Enabling this option will make gtest link the runtimes dynamically too, and match the project in which it is included."
+        set(gtest_force_shared_crt OFF CACHE BOOL "" FORCE)
+    endif()
 
-if(TARGET UrlLib AND JSRUNTIMEHOST_POLYFILL_XMLHTTPREQUEST)
-    set_property(TARGET UrlLib PROPERTY FOLDER Dependencies)
-endif()
+    FetchContent_MakeAvailable_With_Message(googletest)
 
-if(NAPI_JAVASCRIPT_ENGINE STREQUAL "V8" AND JSRUNTIMEHOST_CORE_APPRUNTIME_V8_INSPECTOR)
-    if(NOT TARGET asio)
-        add_library(asio INTERFACE)
-        target_include_directories(asio INTERFACE "${asio_SOURCE_DIR}/asio/include")
-        set_property(TARGET asio PROPERTY FOLDER Dependencies)
-    endif()
-    if(NOT TARGET llhttp_static)
-        set_property(TARGET llhttp_static PROPERTY FOLDER Dependencies)
-    endif()
+    set_property(TARGET gmock PROPERTY FOLDER Dependencies/GoogleTest)
+    set_property(TARGET gmock_main PROPERTY FOLDER Dependencies/GoogleTest)
+    set_property(TARGET gtest PROPERTY FOLDER Dependencies/GoogleTest)
+    set_property(TARGET gtest_main PROPERTY FOLDER Dependencies/GoogleTest)
```

Comment:
I think that Android is also using these so this may not be the right spot for this.

Reply:
I need to fix the condition.

Reply:
I'm not sure why Babylon Native's if condition is different than this.